mypy/semanal.py

```diff
@@ -1527,33 +1527,33 @@ def analyze_property_with_multi_part_definition(
         assert isinstance(first_item, Decorator)
         deleted_items = []
         bare_setter_type = None
+        func_name = first_item.func.name
         for i, item in enumerate(items[1:]):
             if isinstance(item, Decorator):
                 item.func.accept(self)
                 if item.decorators:
                     first_node = item.decorators[0]
-                    if isinstance(first_node, MemberExpr):
+                    if self._is_valid_property_decorator(first_node, func_name):
+                        # Get abstractness from the original definition.
+                        item.func.abstract_status = first_item.func.abstract_status
                         if first_node.name == "setter":
                             # The first item represents the entire property.
                             first_item.var.is_settable_property = True
-                            # Get abstractness from the original definition.
-                            item.func.abstract_status = first_item.func.abstract_status
                             setter_func_type = function_type(
                                 item.func, self.named_type("builtins.function")
                             )
                             assert isinstance(setter_func_type, CallableType)
                             bare_setter_type = setter_func_type
                             defn.setter_index = i + 1
-                        if first_node.name == "deleter":
-                            item.func.abstract_status = first_item.func.abstract_status
                         for other_node in item.decorators[1:]:
                             other_node.accept(self)
                     else:
                         self.fail(
-                            f"Only supported top decorator is @{first_item.func.name}.setter", item
+                            f"Only supported top decorators are @{func_name}.setter and @{func_name}.deleter",
+                            item,
```

Comment:
While we're touching this:

* Could quote the decorators so that they get highlighted in colorized output
* Could use `first_node` as the context, so that `f.setter` gets highlighted in `--pretty` instead of just `f` (I think it should always have line info?)
```suggestion
                            f'Only supported top decorators are "@{func_name}.setter" and "@{func_name}.deleter"',
                            first_node,
```
Up to you if you think those are improvements

Reply:
And this is certainly an improvement! Added an example with `--pretty` and a few other unsupported decorators to tests.